homeassistant/components/august/sensor.py

```diff
@@ -231,10 +230,15 @@ async def async_added_to_hass(self) -> None:
         await super().async_added_to_hass()
 
         last_state = await self.async_get_last_state()
-        if not last_state or last_state.state == STATE_UNAVAILABLE:
+        last_sensor_state = await self.async_get_last_sensor_data()
+        if (
+            not last_state
+            or not last_sensor_state
+            or last_state.state == STATE_UNAVAILABLE
+        ):
             return
 
-        self._attr_native_value = last_state.state
+        self._attr_native_value = last_sensor_state.native_value
```

Comment:
What's the difference between getting the last sensor state from the `get_last_sensor_data` and `get_last_state`?

Reply:
It's explained in the developer docs: https://developers.home-assistant.io/docs/core/entity/sensor#restoring-sensor-states